### PR TITLE
Add `is_partitioned` column 

### LINKED ISSUE #871: [FEATURE]: Add "is_partitioned" field to HMS tables crawler output
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem statement

Special consideration needs to be taken when migrating HMS partitioned tables to UC external table. Currently there is no way to distinguish between HMS partitioned and non-partitioned, based on the output of the assessment.

### Proposed Solution

I see two files tables.py and tables.scala, not sure if both need to be modified or just the scala version.
For the Python version, DESCRIBE EXTENDED outputs partition information if applicable, and could be used to derive is_partitioned.
For the Scala version, the attached piece of code seems to do the trick. I decided not to submit a PR because I am weak in Scala and also this involves a separate function call (getTableMetadata - see the attached notebook) . The notebook also includes two other alternative approaches that did not work.
[test_partitioned.txt](https://github.com/databrickslabs/ucx/files/14131431/test_partitioned.txt)

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by dleiva04**
#  Changes

### Linked issues

Resolves #871 

Related PR:
- https://github.com/databrickslabs/ucx/pull/985

### Functionality 

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [ ] modified existing workflow: `...`
- [ ] added a new table
- [X] modified existing table: `ucx.tables`

### Tests

- [X] manually tested
- [ ] added unit tests
- [ ] added integration tests
- [ ] verified on staging environment (screenshot attached)

image
image

### CODE REVIEW

File: src/databricks/labs/ucx/hive_metastore/tables.scala
Comment (dleiva04) [inline src/databricks/labs/ucx/hive_metastore/tables.scala:1]: Reformat the file for better reading, if this is wrong I can change it back

**Review by nfx — CHANGES_REQUESTED** [review]
please rename the upgrade script to start with `v0.17.0_...` and rebase on the latest `main`. Also create the PR directly on this repo instead of your fork. you can do that by changing the `origin` in `.git/config` file

**Comment by nfx** [comment]
@dleiva04 please rebase/update with the latest released version

**Comment by dleiva04** [comment]
Going to create directly into this repo

**Pull request closed without merging**